### PR TITLE
adds passthrough route for getting ami by amiId

### PR DESCRIPTION
**Pull request by Riturajcse**
https://github.com/Shippable/admiral/issues/969

### CODE REVIEW

File: api/passthrough/getImageByAmiId.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  bag.awsAdapter.EC2.getImages(bag.amiId,
+    function (err, images) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.OperationFailed,
+            'Adapter call failed for EC2.getImages',
+            util.inspect(err))
+        );
+
+      var imgId = _.pluck(images, 'ImageId');
```
Comment (shrivara) [inline api/passthrough/getImageByAmiId.js]: Will this not be an array of image ids?

Comment (shrivara) [inline api/passthrough/getImageByAmiId.js]: Also, what happens when no images are returned?

File: api/passthrough/getImageByAmiId.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+var async = require('async');
+var _ = require('underscore');
+
+var AWSAdapter = require('../../common/awsAdapter.js');
+
+function getImageByAmiId(req, res) {
+  var bag = {
+    reqQuery: req.query,
+    accessToken: req.query.accessToken,
```
Comment (ric03uec) [inline api/passthrough/getImageByAmiId.js]: the tokens are already available in admiral.env file. why are they being passed in request query?

Comment (Riturajcse) [inline api/passthrough/getImageByAmiId.js]: This is the accessToken from the UI that the user enters while enabling dynamic node.

Comment (Riturajcse) [inline api/passthrough/getImageByAmiId.js]: image

**Comment by Riturajcse** [comment]
@ric03uec As discussed I have tested there wont be any issues in calling `updateProvisionSystemIntegration` before `updateSystemMachineImages` in dashboardCtrl.js so I have updated my api passthrough to get access and secret keys from the systemIntegration itself instead of query params. I will send dashboardCtrl changes along with my next changes. As this is a new route and not being called from anywhere so wont be any problem in merging this before updating the dashboardCtrl.

File: api/passthrough/getImageByAmiId.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+      if (!systemIntegrations.length) 
+        return next(
+          new ActErr(who, ActErr.OperationFailed,
+            'No provision systemIntegration found: ')
+        );
+
+      bag.systemIntegration = systemIntegrations[0];
+      if (bag.systemIntegration && bag.systemIntegration.data) {
+        bag.accessToken = bag.systemIntegration.data.accessKey;
```
Comment (ric03uec) [inline api/passthrough/getImageByAmiId.js]: what happens if the keys are not present? will the initialize step fail?

Comment (Riturajcse) [inline api/passthrough/getImageByAmiId.js]: yes this will fail with proper errors.

**Pull request merged**